src/parser/core/modules/racials/dwarf/Stoneform.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import Analyzer from 'parser/core/Analyzer';
+
+import Combatants from 'parser/core/modules/Combatants';
+import { formatNumber } from 'common/format';
+import RACES from 'game/RACES';
+import SpellIcon from 'common/SpellIcon';
+import SPELLS from 'common/SPELLS';
+import StatisticBox from 'interface/others/StatisticBox';
+
+/**
+ * Removes all poison, disease, curse, magic, and bleed effects and reduces all physical damage taken by 10% for 8 sec.
+ *
+ * Example log: https://www.warcraftlogs.com/reports/gYCR2kZaJHVbjfxP/#fight=52&source=16&type=damage-taken
+ */
+
+const STONEFORM_DAMAGE_REDUCTION = 0.1;
+const FALLING_DAMAGE_ABILITY_ID = 3;
+const PHYSICAL_EVENT_TYPE = 1;
+
+class Stoneform extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  
+  damageReduced = 0;
+  physicalDamageTaken = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.race && this.selectedCombatant.race === RACES.Dwarf;
```

Comment:
The first check for `this.selectedCombatant.race` is obsolete since it will always be truthy when it matches `RACES.Dwarf`.